adr/ADR-XX.md

```diff
@@ -0,0 +1,76 @@
+# Subject Mapping Transforms in Streams
+
+|Metadata| Value                     |
+|--------|---------------------------|
+|Date    | 2023-02-10                |
+|Author  | @jnmoyne                  |
+|Status  | `Partially Implemented`   |
+|Tags    | jetstream, client, server |
+
+## Context and Problem Statement
+
+Subject mapping and transformation is only available at the Core NATS level, meaning that in order to define or modify mappings one has to either have access to the server config file, or have access to the account's key in operator security mode. While Core NATS subject mapping has its place and use (e.g. scaling a single stream for writes using partitioning, traffic routing, A/B and Canary testing), many (most) use cases for subject mapping happen in the context of streams, and having to go to the Core NATS server/account level to define subject mappings is quite limiting as it's not easy for an application programmer to be able to define the mappings he/she needs (even if they have access to the account's key).
```

Comment:
I think you can strike this paragraph entirely, or just shorten to "Subject mapping and transformation at the time of client publish (PUB) is useful for many use cases; but there are use cases in stream sourcing (or chaining) where transformation features can also be valuable."

Reply:
I would rather keep it as it is now because IMHO it gives actual examples of what you would use Core NATS subject mapping for, and that it is not easy for an application programmer to define his/her own mappings with Core NATS subject mapping as it's an admin operations. Not sure the paragraph really benefits from being shorten to just that one sentence...